engine/src/main/scala/cromwell/engine/workflow/lifecycle/execution/job/preparation/SubWorkflowPreparationActor.scala

```diff
@@ -11,33 +12,65 @@ import cromwell.engine.workflow.lifecycle.execution.job.preparation.CallPreparat
 import cromwell.engine.{EngineWorkflowDescriptor, WdlFunctions}
 import common.exception.MessageAggregation
-import wom.values.WomEvaluatedCallInputs
+import common.validation.ErrorOr.ErrorOr
+import wom.graph.{GraphInputNode, GraphNode, OptionalGraphInputNode, OptionalGraphInputNodeWithDefault}
+import wom.graph.GraphNodePort.OutputPort
+import wom.values.{WomEvaluatedCallInputs, WomValue}
 
 class SubWorkflowPreparationActor(workflowDescriptor: EngineWorkflowDescriptor,
                                   expressionLanguageFunctions: WdlFunctions,
                                   callKey: SubWorkflowKey,
                                   subWorkflowId: WorkflowId) extends Actor with WorkflowLogging {
 
-  lazy val workflowIdForLogging = workflowDescriptor.id
+  override lazy val workflowIdForLogging = workflowDescriptor.id
 
-  def prepareExecutionActor(inputEvaluation: WomEvaluatedCallInputs): CallPreparationActorResponse = {
+  private def prepareExecutionActor(inputEvaluation: WomEvaluatedCallInputs): ErrorOr[CallPreparationActorResponse] = {
     val oldBackendDescriptor = workflowDescriptor.backendDescriptor
 
-    val newBackendDescriptor = oldBackendDescriptor.copy(
-      id = subWorkflowId,
-      callable = callKey.node.callable,
-      // TODO WOM: need FQN for input definitions somehow ? For now don't worry about subWF
-      knownValues = Map.empty,//workflowDescriptor.knownValues ++ (inputEvaluation map { case (k, v) => k.fullyQualifiedName -> v }),
-      breadCrumbs = oldBackendDescriptor.breadCrumbs :+ BackendJobBreadCrumb(workflowDescriptor.callable, workflowDescriptor.id, callKey)
-    )
-    val engineDescriptor = workflowDescriptor.copy(backendDescriptor = newBackendDescriptor, parentWorkflow = Option(workflowDescriptor))
-    SubWorkflowPreparationSucceeded(engineDescriptor, inputEvaluation)
+    evaluateStartingKnownValues(inputEvaluation, callKey.node.callable.graph.inputNodes) map { startingValues =>
+      val newBackendDescriptor = oldBackendDescriptor.copy(
+        id = subWorkflowId,
+        callable = callKey.node.callable,
+        knownValues = startingValues,
+        breadCrumbs = oldBackendDescriptor.breadCrumbs :+ BackendJobBreadCrumb(workflowDescriptor.callable, workflowDescriptor.id, callKey)
+      )
+      val engineDescriptor = workflowDescriptor.copy(backendDescriptor = newBackendDescriptor, parentWorkflow = Option(workflowDescriptor))
+      SubWorkflowPreparationSucceeded(engineDescriptor, inputEvaluation)
+    }
+  }
+
+  /**
+    * Work out a set of  "workflow inputs" to pass to this subworkflow as though it were a top-level workflow receiving inputs
+    */
+  private def evaluateStartingKnownValues(inputEvaluation: WomEvaluatedCallInputs, workflowInputs: Set[GraphInputNode]): ErrorOr[Map[OutputPort, WomValue]] = {
+
+    /**
+      * Find the values in the provided inputs that match up with subworkflow inputs. Silently drop the rest on the floor.
+      */
```

Comment:
should not be scaladoc, doc generation tools will error with "unmoored comment" messages